dispatch/tests/test_upcoming_event_widget.py

```diff
@@ -0,0 +1,166 @@
+import datetime
+
+from django.template import loader
+
+from rest_framework import status
+
+from dispatch.theme import register
+from dispatch.theme.widgets import Zone, Widget
+from dispatch.theme.event_widget import UpcomingEventsWidget, Sidebar
+from dispatch.theme.fields import CharField, TextField, ArticleField, ImageField, EventField, InvalidField
+from dispatch.apps.content.models import Article, Image
+from dispatch.apps.events.models import Event
+
+from dispatch.tests.cases import DispatchAPITestCase, DispatchMediaTestMixin
+from dispatch.tests.helpers import DispatchTestHelpers
+
+
+class UpcomingEventsTestCase(DispatchAPITestCase):
+
+    def test_upcoming_events(self):
+        """Confirm that the widget can be created, and that we can pull data from it"""
+
+        widget = DispatchTestHelpers.create_upcoming_event_widget(self.client)
+
+        widget_json = widget.to_json()
+
+        self.assertEqual(widget_json['featured_event']['category'], 'academic')
+        self.assertEqual(widget_json['featured_event']['host'], u'test host')
+        self.assertEqual(widget_json['title'], 'Upcoming Events!')
+
+    def test_widget_context(self):
+        """Test the custom context method of the widget"""
+
+        widget = DispatchTestHelpers.create_upcoming_event_widget(self.client)
+
+        widget_data = widget.context(widget.prepare_data())
+
+        self.assertEqual(widget_data['title'], 'Upcoming Events!')
+        self.assertEqual(widget_data['featured_event'].title, u'Title 1')
+        self.assertEqual(widget_data['events'][0]['title'], u'Title 2')
+        self.assertEqual(widget_data['events'][1]['title'], u'Title 3')
+        self.assertEqual(widget_data['events'][2]['title'], u'Title 4')
+        self.assertEqual(widget_data['events'][3]['title'], u'Title 5')
+
+    def test_widget_render(self):
+        """Test the rendering of the event widget"""
+
+        def days_in_future(days):
+            """returns datetime object `days` in the future"""
+            return (datetime.datetime.now() + datetime.timedelta(days=days)).strftime("%B %d, %Y, %I:%M %p").replace('AM', 'a.m.').replace('PM', 'p.m.').replace(" 0", " ")
```

Comment:
What happens if you just call `str(datetime.datetime.now() + datetime.timedelta(days=days)))`?